modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -77,11 +77,11 @@ def get(self):
             The object from the Plasma store.
         """
         logger = get_logger()
-        logger.debug(f"ENTER::Partition.get::{self._identity}")
+        self._is_debug and logger.debug(f"ENTER::Partition.get::{self._identity}")
```

Comment:
let's compute `self._identity` on demand, otherwise we're risking to fail in this scenario:
```python
import modin.config as cfg

part = PandasOnRayPartition(...)
getattr(part, "_identity", None) # None

cfg.LogMode.enable()
part.get() # fails to retrieve ._identity
```

Reply:
It is possible to cache the result of `_is_debug` function after the first call, if we are satisfied that when the new logging mode is enabled, the changes will only be propagated to new partitions.

It seems to me that the situation where new changes affect old objects is more error prone.

Reply:
I think we have to propagate the logger changes to old partitions, consider this example where we want to profile an exact method of an already constructed frame:
```python
df = pd.read_csv(...)

LogMode.enable()
df.sum() # fails
```

Reply:
Ok, it should work now